File: view-mode-sample/src/androidTest/java/com/jaynewstrom/viewmodesample/MainActivityTest.java
package com.jaynewstrom.viewmodesample;

import android.content.pm.ActivityInfo;
import android.content.res.Configuration;
import android.support.test.InstrumentationRegistry;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static android.support.test.espresso.Espresso.onView;
import static android.support.test.espresso.action.ViewActions.click;
import static android.support.test.espresso.assertion.ViewAssertions.matches;
import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
import static android.support.test.espresso.matcher.ViewMatchers.withId;
import static android.support.test.espresso.matcher.ViewMatchers.withText;

@RunWith(AndroidJUnit4.class)
public final class MainActivityTest {

    @Rule public ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class);

    @Test public void ensureContentViewShowsByDefault() {
        onView(withId(R.id.button)).check(matches(isDisplayed()));
    }

    @Test public void ensureContentViewCanBeSwitched() {
        onView(withId(R.id.button)).perform(click());
        onView(withText(R.string.failed_to_load)).check(matches(isDisplayed()));
    }

    @Test public void onRotationEnsureViewModeIsPersisted() {
        onView(withId(R.id.button)).check(matches(isDisplayed()));
        changeOrientation();
        onView(withId(R.id.button)).perform(click());
        onView(withText(R.string.failed_to_load)).check(matches(isDisplayed()));
        changeOrientation();
        onView(withText(R.string.failed_to_load)).check(matches(isDisplayed()));
    }

    private void changeOrientation() {
        int orientation = InstrumentationRegistry.getTargetContext().getResources().getConfiguration().orientation;
        activityRule.getActivity().setRequestedOrientation((orientation == Configuration.ORIENTATION_PORTRAIT)
                ? ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE : ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
    }
}

File: view-mode-sample/src/androidTest/java/com/jaynewstrom/viewmodesample/MainActivityTest.kt
package com.jaynewstrom.viewmodesample

import android.content.Context
import android.content.pm.ActivityInfo
import android.content.res.Configuration
import androidx.test.core.app.ApplicationProvider
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.rule.ActivityTestRule
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class MainActivityTest {
    @get:Rule var activityRule = ActivityTestRule(MainActivity::class.java)

    @Test fun ensureContentViewShowsByDefault() {
        onView(withId(R.id.button)).check(matches(isDisplayed()))
    }

    @Test fun ensureContentViewCanBeSwitched() {
        onView(withId(R.id.button)).perform(click())
        onView(withText(R.string.failed_to_load)).check(matches(isDisplayed()))
    }

    @Test fun onRotationEnsureViewModeIsPersisted() {
        onView(withId(R.id.button)).check(matches(isDisplayed()))
        changeOrientation()
        onView(withId(R.id.button)).perform(click())
        onView(withText(R.string.failed_to_load)).check(matches(isDisplayed()))
        changeOrientation()
        onView(withText(R.string.failed_to_load)).check(matches(isDisplayed()))
    }

    private fun changeOrientation() {
        val orientation = ApplicationProvider.getApplicationContext<Context>().resources.configuration.orientation
        activityRule.activity.requestedOrientation = if (orientation == Configuration.ORIENTATION_PORTRAIT) {
            ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE
        } else {
            ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
        }
    }
}
